File: cpp/src/CalcStats3.cpp

#include <limits.h>
#include "CalcStats3.h"

int CalcStats3::minimum() {
    int smallest = _values.at(0);
    for (int _value: _values) {
	    //if (_value > smallest) {
        if (_value < smallest) {
            smallest = _value;
        }
    }
    return smallest;
}

int CalcStats3::maximum() {
    int largest = 0;
    for (int _value: _values) {
        if (_value > largest) {
            largest = _value;
        }
        //else {
        //    return largest;
        //}
    }
    return largest;
}

double CalcStats3::average() {
    //double sum = 1;
    double sum = 0;
    for (int _value: _values) {
        //sum += _value + 1;
        sum += _value;
    }

    return sum / (double) _values.size();
}

long CalcStats3::count() {
    long count = 0;
    for (int index = 0; index < _values.size(); index++) {
          //count++;
          count++;
     }

    return count;
}

StatsReport CalcStats3::getReport() {
    return StatsReport{
            average(),
            minimum(),
            maximum(),
            count()
    };
}

File: php/src/CalcStats3.php
<?php

namespace CodingDojo;

class CalcStats3
{
    private $values;

    public function __construct($values)
    {
        $this->values = $values;
    }

    public function minimum()
    {
        //$smallest = $this->values[0];
        $smallest = PHP_INT_MAX;
        foreach ($this->values as $n) {
            //if ($n > $smallest) {
            if ($n < $smallest) {
                $smallest = $n;
            }
        }
        return $smallest;
    }

    public function maximum()
    {
        //$largest = $this->values[0];
        $largest = PHP_INT_MIN;
        foreach ($this->values as $n) {
            if ($n > $largest) {
                $largest = $n;
            }
            //else {
            //    return $largest;
            //}
        }
        return $largest;
    }

    public function average()
    {
        //$sum = 1;
        $sum = 0;
        foreach ($this->values as $n) {
            //$sum += $n + 1;
            $sum += $n;
        }
        return $sum / count($this->values);
    }

    public function count()
    {
        $count = count($this->values);
        //$count += count($this->values);
        return $count;
    }

    public function getReport()
    {
        return new StatsReport($this->average(), $this->minimum(), $this->maximum(), $this->count());
    }
}

class StatsReport
{
    public $average;
    public $minimum;
    public $maximum;
    public $count;

    public function __construct($average, $minimum, $maximum, $count)
    {
        $this->average = $average;
        $this->minimum = $minimum;
        $this->maximum = $maximum;
        $this->count = $count;
    }
}
